### PR TITLE
CI improvements

### LINKED ISSUE #174: check if wasm build is up to date
the CI should check if the binary of all template are the latest version

### LINKED ISSUE #250: fix miri run
consider using https://github.com/rust-lang/miri and possibly integrating in the CI

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201757896515792/1201762933103389) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by Fraccaman**
- [x] Integrate Miri (close #250)
- [x] Check if wasm build are up to date. If not, commit them back. Use the same docker image provided in the source code.(close #174)

### CODE REVIEW

**Review by tzemanovic — APPROVED** [review]
LGTM, thanks! If you could pls move just move the sh scripts in the ci dir before merging

>

**Comment by Fraccaman** [comment]
@tzemanovic ready to be merged! The CI is using the new image

**Pull request merged**
